src/terms/operators.jl

```diff
@@ -111,16 +111,16 @@ Matrix(op::NonlocalOperator) = op.P * op.D * op.P'
 """
 Magnetic field operator A⋅(-i∇).
 """
-struct MagneticFieldOperator <: RealFourierOperator
-    basis
-    kpoint
-    Apot  # Apot[α][i,j,k] is the A field in direction α
+struct MagneticFieldOperator{T <: Real} <: RealFourierOperator
+    basis::PlaneWaveBasis{T}
+    kpoint::Kpoint{T}
+    Apot  # Apot[α][i,j,k] is the A field in (cartesian) direction α
 end
 @timing_seq "apply MagneticFieldOperator" function apply!(Hψ, op::MagneticFieldOperator, ψ)
     # TODO this could probably be better optimized
     for α = 1:3
         all(op.Apot[α] .== 0) && continue
-        pα = [(G[α] + op.kpoint.coordinate[α]) for G in G_vectors(op.kpoint)]
+        pα = [(op.basis.model.recip_lattice*(G + op.kpoint.coordinate))[α] for G in G_vectors(op.kpoint)]
```

Comment:
Add a space around `*` and line too long.

Reply:
Also will be taken care of by the _cart

Reply:
same as above.